### PR TITLE
Bug 1273983 - Allow enabling/disabling a performance framework

### PR DESCRIPTION
**Pull request by SJasoria**
Here are the changes without the pytests.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1620)

### CODE REVIEW

File: treeherder/etl/perf.py [hunk 1]
```diff
@@ -95,6 +95,9 @@ def _load_perf_artifact(project_name, reference_data, job_data, job_guid,
                        "load of performance artifacts".format(
                            perf_datum['framework']['name']))
         return
+    if perf_datum['framework']['enabled'] is False:
```
Comment (wlach) [inline treeherder/etl/perf.py]: This isn't the right place to look for enabled. You want the framework object that's a few lines above.

**Comment by wlach** [comment]
In order for the tests to pass, uou need to add a migration for this change by running:

```
./manage.py makemigrations -n performance_framework_add_enabled_field
```

Check in the file that running this command creates.

I'd focus on getting the existing tests to pass before writing new ones.

File: treeherder/perf/models.py [hunk 2]
```diff
@@ -15,13 +15,17 @@
...
     name = models.SlugField(max_length=255, unique=True)
+    enabled = models.BooleanField(default=False)
 
     class Meta:
         db_table = 'performance_framework'
 
     def __str__(self):
         return self.name
 
+    def __getitem__(self, enabled):
```
Comment (wlach) [inline treeherder/perf/models.py]: What's the purpose of this?

File: treeherder/etl/perf.py [hunk 3]
```diff
@@ -95,6 +95,10 @@ def _load_perf_artifact(project_name, reference_data, job_data, job_guid,
                        "load of performance artifacts".format(
                            perf_datum['framework']['name']))
         return
+    if framework['enabled'] is False:
+        logger.warning("Performance framework {} is not enabled, skipping"
```
Comment (wlach) [inline treeherder/etl/perf.py]: This should be info rather than warning

File: treeherder/perf/models.py (hunk 2 above)
Comment (SJasoria) [inline treeherder/perf/models.py]: It allows to access 'enabled' field in the framework. If it is not included the following error pops up while running the tests:

File "/home/vagrant/treeherder/treeherder/etl/perf.py", line 98, in _load_perf_artifact
    if framework['enabled'] is False:
TypeError: 'PerformanceFramework' object has no attribute '__getitem__'

Comment (wlach) [inline treeherder/perf/models.py]: Oh! You shouldn't access 'enabled' that way. Do it this way:

```py
if not framework.enabled:
```

Then you can take this out. :)

**Comment by wlach** [comment]
This is looking good except for the comment. 

I'd also like you to enable editing the PerformanceFramework entries in the admin UI. I believe you should be able to do this by adding PerformanceFramework to browserid_admin here:

treeherder/treeherder/webapp/admin.py

You can test your changes by logging into http://local.treeherder.mozilla.org/admin as a user with superuser permissions (you can then add these permissions by following these instructions: http://stackoverflow.com/a/11337600)

**Pull request merged**